### DISCUSSION TITLE
Should nethost be migrtated from coreclr to nethost and hostfxr libraries?

### DISCUSSION THREAD

**Discussion by MAAC-2018** [dotnet/runtime#88388] [Q&A] [answered]
I currently have a coreclr host and that interops with NET6.0.18 without any issue.
Microsoft site indicates that coreclr is obsolete and any host using 3.0 or greater should use nethost and hostfxr
https://learn.microsoft.com/en-us/dotnet/core/tutorials/netcore-hosting
Is there really a need to migrate?

**Comment by teo-tsirpanis** [comment]
c.c. @vitek-karas

**Comment by vitek-karas** [comment, marked as answer]
We consider the `coreclr.dll` to be an internal implementation detail. We do know that it's used by some hosts, but we actively discourage people from doing that. And we may decide to break it in the future - for example https://github.com/dotnet/runtime/issues/52688 could break your host (depends on how you're using coreclr) if we decide to do it.

There are also functional reasons for this - for example `AssemblyDependencyResolver` will not work if the runtime is hosted without `hostfxr`/`hostpolicy`.

So in short I would suggest you strongly consider moving your code to the `hostfxr` APIs. It should also simplify your code as hosting the runtime directly usually needs more setup code.

**Reply by MAAC-2018** [reply]
Thanks, for the feedback, much appreciated.
